crates/holochain/tests/tests/publish/mod.rs

```diff
@@ -2,11 +2,13 @@ use holo_hash::ActionHash;
 use holochain::core::workflow::publish_dht_ops_workflow::num_still_needing_publish;
 use holochain::sweettest::*;
 use holochain_wasm_test_utils::TestWasm;
+use holochain_zome_types::prelude::GetValidationReceiptsInput;
+use holochain_zome_types::validate::ValidationReceiptSet;
 
 /// Verifies that publishing terminates naturally when enough validation receipts are received.
 #[cfg(feature = "test_utils")]
 #[tokio::test(flavor = "multi_thread")]
-#[ignore = "receipt completion is flaky, revise once integration logic is merged into app validation workflow"]
+#[ignore = "receipt completion is flaky, which has been rechecked since the workflow reviews"]
```

Comment:
This is still flaky, and I think that's a cause for concern really. Possibly related to not driving app validation forwards aggressively enough? I'm not sure. It seems much more stable now than it was before and the test passes fine for me locally some of the time. Not reliably though.

Reply:
Hopefully putting this block in a retry loop would solve it https://github.com/holochain/holochain/blob/bcdc6711ce3bd130503582294e02b99ee21285be/crates/holochain/tests/tests/publish/mod.rs#L64-L74

After all, above it says:

> // Wait until they all see the created entry, at that point validation receipts should be getting sent soon

emphasis on "should"

Reply:
Yes good point, that would make that part of the test more reliable because the receipts haven't definitely been received by then. However, I tend to see the failure happen at the timeout step with `timed out waiting for all receipts to be received: Elapsed(())`